ludwig/hyperopt/run.py

```diff
@@ -373,3 +404,34 @@ def update_hyperopt_params_with_defaults(hyperopt_params):
         hyperopt_params[EXECUTOR],
         executor_defaults,
     )
+
+
+def get_shared_params_dict(config_feature_type: str, features: Dict[str, Any]) -> Dict[str, Set]:
+    """Generates a mapping of feature type to the corresponding set of features without an encoder or one using the
+    default encoder for that feature type.
```

Comment:
@justinxzhao yes, we currently don't support hyperopt on preprocessing parameters as it's too expensive. It would be good to clarify this in the docstring as a comment.

@arnavgarg1 do we now support hyperopt search over output features? If we do, it would be good to update that in the documentation.

Reply:
@justinxzhao The rationale is primarily that using default encoders makes it easier to ensure that the shared parameters being defined are actually applicable to all features of that type because they all share the same parameters. Different encoders/decoders are likely to have different parameters so this was just a simplification. Theoretically, we can add another layer of complexity to do defaults per feature type per encoder, but I don't think we need to provide this level of granularity at this time. Let me know if you feel differently!

@ShreyaR will add both things to the documentation/doc string.